lib/rules/simplify-set-operations.ts

```diff
@@ -0,0 +1,419 @@
+             * - `[^[^a]&&[^b]]` -> `[ab]`
+             * - `[[^a]&&[^b]&&c]` -> `[[^ab]&&c]`
+             */
+            function toNegationOfDisjunction(
+                eccNode: ExpressionCharacterClass,
+            ) {
+                const expression = eccNode.expression
+                if (expression.type !== "ClassIntersection") {
+                    return false
+                }
+                const operands = collectIntersectionOperands(expression)
+                const negateOperands: (NegateCharacterClassElement &
+                    ClassSetOperand)[] = []
+                const others: ClassSetOperand[] = []
+                for (const e of operands) {
+                    if (isNegate(e)) {
+                        negateOperands.push(e)
+                    } else {
+                        others.push(e)
+                    }
+                }
+                const fixedOperands = negateOperands
+                    .map((negateOperand) => getRawTextToNot(negateOperand))
+                    .join("")
+                if (negateOperands.length === operands.length) {
+                    return reportWhenFixedIsCompatible({
+                        reportNode: eccNode,
+                        targetNode: eccNode,
+                        messageId: "toNegationOfDisjunction",
+                        data: {
+                            target: "character class",
+                        },
+                        fix: () =>
+                            `[${eccNode.negate ? "" : "^"}${fixedOperands}]`,
+                    })
+                }
+                if (negateOperands.length < 2) {
+                    return null
+                }
+                return reportWhenFixedIsCompatible({
+                    reportNode: negateOperands[negateOperands.length - 1]
+                        .parent as ClassIntersection,
+                    targetNode: eccNode,
+                    messageId: "toNegationOfDisjunction",
+                    data: {
+                        target: "expression",
+                    },
+                    fix: () => {
+                        const operandTestList = [
+                            `[^${fixedOperands}]`,
+                            ...others.map((e) => e.raw),
+                        ]
+                        return `[${
+                            eccNode.negate ? "^" : ""
+                        }${operandTestList.join("&&")}]`
+                    },
+                })
+            }
+
+            /**
+             * Checks the given character class and reports if it can be converted to the negation of a conjunction
+             * using De Morgan's laws.
+             * Returns true if reported.
+             *
+             * e.g.
+             * - `[[^a][^b]]` -> `[^a&&b]`
+             */
+            function toNegationOfConjunction(ccNode: CharacterClass) {
+                if (ccNode.elements.length <= 1 || !flags.unicodeSets) {
```

Comment:
I think this flag check is redundant.

```suggestion
                if (ccNode.elements.length <= 1) {
```